indexing-service/src/main/java/io/druid/indexing/worker/WorkerTaskMonitor.java

```diff
@@ -84,14 +89,20 @@ public WorkerTaskMonitor(
   public void start()
   {
     try {
+      // restore restorable tasks
+      final List<Pair<Task, ListenableFuture<TaskStatus>>> restored = taskRunner.restore();
+      for (Pair<Task, ListenableFuture<TaskStatus>> pair : restored) {
+        submitTaskRunnable(pair.lhs, pair.rhs);
```

Comment:
The executor has an unbounded queue, so the tasks would just get added to the queue.